test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java

```diff
@@ -337,16 +337,36 @@ public void handle(final HttpExchange exchange) throws IOException {
                 exchange.getResponseBody().write(response);
 
             } else {
+                logger.error("unknown request: {}", request);
                 exchange.sendResponseHeaders(RestStatus.INTERNAL_SERVER_ERROR.getStatus(), -1);
             }
+        } catch (Exception e) {
+            logger.error("exception in request " + request, e);
+            throw e;
         } finally {
             exchange.close();
         }
     }
 
+    private boolean isUploadPartRequest(String request) {
+        return Regex.simpleMatch("PUT /" + path + "/*?uploadId=*&partNumber=*", request)
+            || Regex.simpleMatch("PUT /" + path + "/*?partNumber=*&uploadId=*", request);
```

Comment:
Could you distinguish V1 vs V2 request formats, and add a todo someplace? I'm imagining that we should clean up and remove the V1 formats after the upgrade.

Reply:
I opened ES-11379. Honestly this is all hideous, we should be parsing the request properly, but it's only a test fixture...